systemtests/src/main/java/io/enmasse/systemtest/selenium/page/OpenshiftWebPage.java

```diff
@@ -0,0 +1,383 @@
+
+    private WebElement getCatalog() {
+        return selenium.getDriver().findElement(By.className("services-no-sub-categories"));
+    }
+
+    private String getTitleFromService(WebElement element) {
+        return element.findElement(By.className("services-item-name")).getAttribute("title");
+    }
+
+    public WebElement getServiceFromCatalog(String name) {
+        List<WebElement> services = getServicesFromCatalog();
+        return services.stream().filter(item -> name.equals(getTitleFromService(item))).collect(Collectors.toList()).get(0);
+    }
+
+    public List<WebElement> getServicesFromCatalog() {
+        List<WebElement> services = getCatalog().findElements(By.className("services-item"));
+        services.forEach(item -> log.info("Got service item from catalog: {}", getTitleFromService(item)));
+        return services;
+    }
+
+    private WebElement getModalWindow() {
+        return selenium.getDriver().findElement(By.className("modal-content"));
+    }
+
+    private WebElement getNextButton() {
+        return getModalWindow().findElement(By.id("nextButton"));
+    }
+
+    private WebElement getBackButton() {
+        return getModalWindow().findElement(By.id("backButton"));
+    }
+
+    private WebElement getAddToProjectDropDown() {
+        return getModalWindow().findElement(By.className("dropdown"));
+    }
+
+    private WebElement getItemFromAddToProjectDropDown(String projectName) {
+        List<WebElement> result = getItemsFromAddToProjectDropDown().stream()
+                .filter(item -> projectName.equals(getTextFromAddToProjectDropDownItem(item))).collect(Collectors.toList());
+        return result.size() > 0 ? result.get(0) : null;
+    }
+
+    private List<WebElement> getItemsFromAddToProjectDropDown() {
+        List<WebElement> dropdownItems = getAddToProjectDropDown()
+                .findElement(By.className("dropdown-menu")).findElements(By.className("ui-select-choices-row"));
+        for (WebElement el : dropdownItems) {
+            log.info("Got add to project choice: {}", getTextFromAddToProjectDropDownItem(el));
+        }
+        return dropdownItems;
+    }
+
+    private String getTextFromAddToProjectDropDownItem(WebElement item) {
+        return item.findElement(By.className("ui-select-choices-row-inner")).findElement(By.tagName("span")).getText();
+    }
+
+    private List<WebElement> getBindingRadioItems() {
+        return getModalWindow().findElement(By.className("radio")).findElements(By.tagName("label"));
+    }
+
+    private WebElement getProjectListItem(String name) {
+        selenium.takeScreenShot();
+        List<WebElement> projects = selenium.getDriver().findElements(By.className("list-group-item"));
+        for (WebElement el : projects) {
+            if (el.findElement(By.tagName("span")).getText().equals(name)) {
+                return el;
+            }
+        }
+        return null;
```

Comment:
Exception should be better

Reply:
there is ok with null, parent method handle null state exception.